src/daemon/daemon.cpp

```diff
@@ -3362,3 +3356,71 @@ void mp::Daemon::reply_msg(grpc::ServerReaderWriterInterface<Reply, Request>* se
 
     server->Write(reply);
 }
+
+void mp::Daemon::populate_instance_info(VirtualMachine& vm, mp::DetailedInfoItem* info, bool runtime_info, bool deleted,
+                                        bool& have_mounts)
+{
+    const auto& name = vm.vm_name;
+    auto instance_info = info->mutable_instance_info();
+    auto present_state = vm.current_state();
+    info->set_name(name);
+    if (deleted)
+        info->mutable_instance_status()->set_status(mp::InstanceStatus::DELETED);
+    else
+        info->mutable_instance_status()->set_status(grpc_instance_status_for(present_state));
+
+    auto vm_image = fetch_image_for(name, config->factory->fetch_type(), *config->vault);
+    auto original_release = vm_image.original_release;
+
+    if (!vm_image.id.empty() && original_release.empty())
+    {
+        try
+        {
+            auto vm_image_info = config->image_hosts.back()->info_for_full_hash(vm_image.id);
+            original_release = vm_image_info.release_title.toStdString();
+        }
+        catch (const std::exception& e)
+        {
+            mpl::log(mpl::Level::warning, category, fmt::format("Cannot fetch image information: {}", e.what()));
+        }
+    }
+
+    instance_info->set_num_snapshots(vm.get_num_snapshots());
+    instance_info->set_image_release(original_release);
+    instance_info->set_id(vm_image.id);
+
+    auto vm_specs = vm_instance_specs[name];
+
+    auto mount_info = info->mutable_mount_info();
+    populate_mount_info(vm_specs.mounts, mount_info, have_mounts);
+
+    if (!runtime_info && mp::utils::is_running(present_state))
```

Comment:
`runtime_info` is the opposite of what this parameter currently means, isn't it? 

I agree it would make for sense to drop the negative, but then the logic needs inverting too, i.e. drop the `!` here and add it where you pass the argument from the grpc request.